Tests/MongoSwiftSyncTests/UnifiedTestRunner/UnifiedTestRunner.swift

```diff
@@ -183,9 +183,13 @@ struct UnifiedTestRunner {
                     let collEntities = context.entities.values.compactMap { try? $0.asCollection() }
                     for (_, client) in try self.internalClient.asMongosClients() {
                         for entity in collEntities {
-                            _ = try client.db(entity.namespace.db).runCommand(
-                                ["distinct": .string(entity.name), "key": "_id"]
-                            )
+                            do {
+                                _ = try client.db(entity.namespace.db).runCommand(
+                                    ["distinct": .string(entity.name), "key": "_id"]
+                                )
+                            } catch let commandError as MongoError.CommandError where commandError.code == 26 {
```

Comment:
I noticed that after I synced in the unified change stream spec tests, they started to fail on macOS + sharded clusters, with a similar NamespaceNotFound error problem to that Rohan was encountering. 
I think this was likely also an issue around the exact namespaces that exist / don't exist at the time those tests are run due to test ordering